### PR TITLE
Freeze FlaxWav2Vec2 Feature Encoder

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
This PR enables the FlaxWav2Vec2 Feature Encoder to be _frozen_. The flow of gradients through the Feature Encoder during forward or reverse-mode automatic differentiation is prevented through use of a simple command (`jax.lax.stop_gradient`), thus holding the Feature Encoder parameters fixed. Freezing the Feature Encoder is required for the majority of FlaxWav2Vec2 fine-tuning set-ups, making this PR an important addition to the FlaxWav2Vec2Model.

### CODE REVIEW

**Review by patrickvonplaten — COMMENTED** [review]
Looks good to me! Can we maybe add one small test ? E.g. backprop a loss and just make sure the gradients stay the same

**Review by patrickvonplaten — APPROVED** [review]

**Comment by patrickvonplaten** [comment]
As discussed offline - the test didn't check what it was supposed to check. Bad review from my part. We should open a new PR here to fix it cc @patil-suraj (sorry should have put you as a reviewer as well)
